lib/resque.rb

```diff
@@ -147,15 +167,25 @@ def inline=(inline)
   #
   # Returns nothing
   def push(queue, item)
-    watch_queue(queue)
-    redis.rpush "queue:#{queue}", encode(item)
+    redis.pipelined do
+      watch_queue(queue)
+      redis.rpush "queue:#{queue}", encode(item)
+    end
   end
 
-  # Pops a job off a queue. Queue name should be a string.
+  # Pops a job off one ore more queues.
   #
-  # Returns a Ruby object.
-  def pop(queue)
-    decode redis.lpop("queue:#{queue}")
+  # queues -  An array of String queue names.
+  # timeout - Integer number of seconds to block on BLPOP.
+  #           Defaults to 1, requires a minimum value of 1 to prevent
+  #           indeterminate blocking.
+  #
+  # Returns an array of [queue_name, decoded_payload], or falsey.
+  def pop(queues, timeout=1)
```

Comment:
Tolerates calls with the standard signature (scalar `queue`) but different return type from the built-in